File: models/payment.py
from . import db
import uuid
from datetime import datetime


class Payment(db.Model):
    __tablename__ = 'payments'

    id = db.Column(db.Integer, primary_key=True)
    uuid = db.Column(db.String(36), unique=True, nullable=False, default=lambda: str(uuid.uuid4()))
    doador_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    instituicao_id = db.Column(db.Integer, db.ForeignKey('institution_profiles.id'), nullable=False)
    necessidade_id = db.Column(db.Integer, db.ForeignKey('needs.id'), nullable=True)
    valor = db.Column(db.Numeric(10, 2), nullable=False)
    metodo = db.Column(db.String(20), nullable=False)
    status = db.Column(db.String(20), default='pendente')
    comprovante_url = db.Column(db.String(500))
    transacao_id = db.Column(db.String(100))
    data_criacao = db.Column(db.DateTime, default=db.func.now())
    data_confirmacao = db.Column(db.DateTime)

    doador = db.relationship('User', backref='pagamentos', foreign_keys=[doador_id])
    instituicao = db.relationship('InstitutionProfile', backref='pagamentos', foreign_keys=[instituicao_id])
    necessidade = db.relationship('Need', backref='pagamentos', foreign_keys=[necessidade_id])

    STATUS_CHOICES = ('pendente', 'processando', 'confirmado', 'recusado', 'cancelado')
    METODO_CHOICES = ('pix', 'cartao_credito', 'boleto', 'transferencia')

File: src/main/java/com/conexoessolidarias/model/Payment.java
package com.conexoessolidarias.model;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "payments")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false, length = 36)
    private String uuid = UUID.randomUUID().toString();

    @ManyToOne
    @JoinColumn(name = "doador_id", nullable = false)
    private User doador;

    @ManyToOne
    @JoinColumn(name = "instituicao_id", nullable = false)
    private InstitutionProfile instituicao;

    @ManyToOne
    @JoinColumn(name = "necessidade_id")
    private Need necessidade;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal valor;

    @Column(nullable = false, length = 20)
    private String metodo;

    @Column(nullable = false)
    private String status = "pendente";

    @Column(length = 500)
    private String comprovanteUrl;

    @Column(length = 100)
    private String transacaoId;

    private LocalDateTime dataCriacao = LocalDateTime.now();

    private LocalDateTime dataConfirmacao;

    public Payment() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getUuid() { return uuid; }
    public void setUuid(String uuid) { this.uuid = uuid; }
    public User getDoador() { return doador; }
    public void setDoador(User doador) { this.doador = doador; }
    public InstitutionProfile getInstituicao() { return instituicao; }
    public void setInstituicao(InstitutionProfile instituicao) { this.instituicao = instituicao; }
    public Need getNecessidade() { return necessidade; }
    public void setNecessidade(Need necessidade) { this.necessidade = necessidade; }
    public BigDecimal getValor() { return valor; }
    public void setValor(BigDecimal valor) { this.valor = valor; }
    public String getMetodo() { return metodo; }
    public void setMetodo(String metodo) { this.metodo = metodo; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getComprovanteUrl() { return comprovanteUrl; }
    public void setComprovanteUrl(String comprovanteUrl) { this.comprovanteUrl = comprovanteUrl; }
    public String getTransacaoId() { return transacaoId; }
    public void setTransacaoId(String transacaoId) { this.transacaoId = transacaoId; }
    public LocalDateTime getDataCriacao() { return dataCriacao; }
    public void setDataCriacao(LocalDateTime dataCriacao) { this.dataCriacao = dataCriacao; }
    public LocalDateTime getDataConfirmacao() { return dataConfirmacao; }
    public void setDataConfirmacao(LocalDateTime dataConfirmacao) { this.dataConfirmacao = dataConfirmacao; }
}
